ext/standard/var.c

```diff
@@ -560,27 +569,40 @@ PHPAPI void php_var_export_ex(zval *struc, int level, smart_str *buf) /* {{{ */
 				buffer_append_spaces(buf, level - 1);
 			}
 
+			zend_class_entry *ce = Z_OBJCE_P(struc);
+			bool is_enum = ce->ce_flags & ZEND_ACC_ENUM;
+
 			/* stdClass has no __set_state method, but can be casted to */
-			if (Z_OBJCE_P(struc) == zend_standard_class_def) {
+			if (ce == zend_standard_class_def) {
 				smart_str_appendl(buf, "(object) array(\n", 16);
 			} else {
-				smart_str_append(buf, Z_OBJCE_P(struc)->name);
-				smart_str_appendl(buf, "::__set_state(array(\n", 21);
+				smart_str_append(buf, ce->name);
+				if (is_enum) {
+					zend_object *zobj = Z_OBJ_P(struc);
+					zval tmp;
+					zval *case_name_zval = zend_read_property_ex(ce, zobj, ZSTR_KNOWN(ZEND_STR_NAME), 0, &tmp);
+					smart_str_appendl(buf, "::", 2);
+					smart_str_append(buf, Z_STR_P(case_name_zval));
+				} else {
+					smart_str_appendl(buf, "::__set_state(array(\n", 21);
+				}
 			}
 
-			if (myht) {
+			if (myht && !is_enum) {
 				ZEND_HASH_FOREACH_KEY_VAL_IND(myht, index, key, val) {
 					php_object_element_export(val, index, key, level, buf);
 				} ZEND_HASH_FOREACH_END();
 				GC_TRY_UNPROTECT_RECURSION(myht);
+			}
+			if (myht) {
 				zend_release_properties(myht);
 			}
 			if (level > 1) {
```

Comment:
The code would be clearer if this returned immediately for enums

```c
if (is_enum) {
    return;
}
```

```suggestion
			if (level > 1 && !is_enum) {
```

This could also include a test of dumping enums in arrays, e.g.

```php
enum Test: string {
    case X = 'x';
}

var_export(Test::cases());
```

Reply:
> The code would be clearer if this returned immediately for enums
> ```c
> if (is_enum) {
>     return;
> }
> ```

I'm pretty sure this change is needed to fix the whitespace generated by var_dump for `var_dump([[Suit::Hearts]]);`